src/azure-cli/azure/cli/command_modules/appconfig/custom.py

```diff
@@ -95,7 +135,8 @@ def update_configstore(cmd,
     update_params = ConfigurationStoreUpdateParameters(tags=tags,
                                                        sku=Sku(name=sku) if sku else None,
                                                        public_network_access=public_network_access,
-                                                       disable_local_auth=disable_local_auth)
+                                                       disable_local_auth=disable_local_auth,
+                                                       enable_purge_protection=enable_purge_protection)
```

Comment:
The way RP handles enable_purge_protection is this:
- default value of enable_purge_protection is false
- update can be called multiple times with enable_purge_protection=false if it is already false (i.e., it wasn't turned on during create)
- update can be called EXACTLY ONCE with enable_purge_protection=true if it was originally false (i.e., it wasn't turned on during create)
- Once enable_purge_protection has been set to true, it cannot accept true or false anymore. RP only accepts None thereafter. If any request is made with true or false, this error is thrown:
```
(InvalidProperty) The property 'EnablePurgeProtection' is not valid.
Code: InvalidProperty
Message: The property 'EnablePurgeProtection' is not valid.
Additional Information:Type: ActivityId
Info: {
    "activityId": "3ddb9188-477c-44be-ac93-3ac5892e02c8"
}
```

This error should not be displayed to users since it doesn't help understand what's wrong. We need to find a way to catch this specific error and display a helpful error message.

Reply:
While we should do proper error handling for this case, I believe we need to rethink how RP handles request with `true` value. It should only error out when the value is `false`.

Reply:
As per the discussion in the scrum, this should be handled by the RP. Joe has a PR out.
We won't need updated SDK because it is an implementation change, not an API change.
We are good to merge this PR.